src/drivers/imu/mpu9250/mpu9250.cpp

```diff
@@ -368,7 +397,7 @@ MPU9250::init()
 #endif /* USE_I2C */
 
 	/* do CDev init for the mag device node, keep it optional */
-	if (_whoami == MPU_WHOAMI_9250) {
+	if ( _whoami == MPU_WHOAMI_9250 || _device_type == MPU_DEVICE_TYPE_ICM20948 ) {
```

Comment:
shouldn't we check _whoami also for the ICM20948 instead of _device_type == MPU_DEVICE_TYPE_ICM20948?

Reply:
Yes, this was a bit provisional to get going and no lasting solution. I cleaned it up and introduced a new device type for the MPU6500, will push later.